src/generic/clipboard/hooks/useClipboard.ts

```diff
@@ -0,0 +1,82 @@
+import { useIntl } from '@edx/frontend-platform/i18n';
+import { useQuery, useQueryClient } from '@tanstack/react-query';
+import { useContext, useEffect, useState } from 'react';
+
+import { getClipboard, updateClipboard } from '../../data/api';
+import {
+  CLIPBOARD_STATUS,
+  STRUCTURAL_XBLOCK_TYPES,
+  STUDIO_CLIPBOARD_CHANNEL,
+} from '../../../constants';
+import { ToastContext } from '../../toast-context';
+import messages from './messages';
+
+/**
+ * Custom React hook for managing clipboard functionality.
+ *
+ * @param canEdit - Flag indicating whether the clipboard is editable.
+ * @returns - An object containing state variables and functions related to clipboard functionality.
+ * @property showPasteUnit - Flag indicating whether the "Paste Unit" button should be visible.
+ * @property showPasteXBlock - Flag indicating whether the "Paste XBlock" button should be visible.
+ * @property sharedClipboardData - The shared clipboard data object.
+ * @property copyToClipboard - Function to copy the current selection to the clipboard.
+ */
+const useClipboard = (canEdit: boolean = true) => {
+  const intl = useIntl();
+  const [clipboardBroadcastChannel] = useState(() => new BroadcastChannel(STUDIO_CLIPBOARD_CHANNEL));
+  const { data: clipboardData } = useQuery({
+    queryKey: ['clipboard'],
+    queryFn: getClipboard,
+    refetchInterval: (data) => (data?.content?.status === CLIPBOARD_STATUS.loading ? 1000 : false),
+  });
+  const { showToast } = useContext(ToastContext);
+
+  const queryClient = useQueryClient();
+
+  const copyToClipboard = async (usageKey: string) => {
+    // This code is synchronous for now, but it could be made asynchronous in the future.
+    // In that case, the `done` message should be shown after the asynchronous operation completes.
+    showToast(intl.formatMessage(messages.copying));
+    try {
+      const newData = await updateClipboard(usageKey);
+      clipboardBroadcastChannel.postMessage(clipboardData);
```

Comment:
```suggestion
      clipboardBroadcastChannel.postMessage(newData);
```
This needs to be `newData`, or it doesn't work properly (shows old things in the clipboard).

Reply:
My bad! Fixed here: c43d2e63af6ac7fec3a1b412df83686282aca42b